packages/veritone-react-common/src/helpers/withMuiThemeProvider.js

```diff
@@ -1,48 +1,68 @@
 import React from 'react';
-import { func } from 'prop-types';
-import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';
-import blue from '@material-ui/core/colors/blue';
-
-export default function withMuiThemeProvider(Component) {
-  class WrappedWithMuiTheme extends React.Component {
-    static displayName = Component.displayName || Component.name;
-    static propTypes = {
-      forwardedRef: func
-    };
-
-    render() {
-      const { forwardedRef, ...rest } = this.props;
-      return (
-        <MuiThemeProvider
-          theme={createMuiTheme({
-            palette: {
-              primary: {
-                light: blue[300],
-                main: blue[500],
-                dark: blue[700]
-              }
-            },
-            typography: {
-              button: {
-                fontWeight: 400
-              }
-            }
-          })}
-        >
-          <Component ref={forwardedRef} {...rest} />
-        </MuiThemeProvider>
-      );
-    }
+import { oneOfType, func, node, objectOf, object, any } from 'prop-types';
+import {
+  MuiThemeProvider,
+  createMuiTheme,
+  jssPreset
+} from '@material-ui/core/styles';
+import { create } from 'jss';
+import JssProvider from 'react-jss/lib/JssProvider';
+
+import { guid } from './guid';
+
+export class VSDKStyleWrapper extends React.PureComponent {
+  static propTypes = {
+    theme: objectOf(any),
+    children: node
+  };
+
+  constructor() {
+    super();
+    this._style = `vsdk-${guid()}`;
+    this._jss = create(jssPreset());
   }
 
-  function forwardRef(props, ref) {
-    return <WrappedWithMuiTheme {...props} forwardedRef={ref} />;
+  render() {
+    const mergedTheme = this.props.theme
```

Comment:
since props.theme will be undefined, you don't need the conditional (also the cloning step is unnecessary). so in all cases

```javascript
 const mergedTheme = createMuiTheme(this.props.theme)
```
should be fine.

also what happened to the default palette? I think we should indeed deep-merge user themes with ours (with the passed-in theme as higher priority).